src/plugins/data/common/index_patterns/expressions/load_index_pattern.ts

```diff
@@ -57,4 +58,24 @@ export const getIndexPatternLoadMeta = (): Omit<
       }),
     },
   },
+  extract(state) {
+    const refName = 'indexPatternLoad.id';
+    const references: SavedObjectReference[] = [
+      {
+        name: refName,
+        type: 'search',
+        id: state.id[0] as string,
+      },
+    ];
+    state.id[0] = refName;
+    return { state, references };
+  },
+
+  inject(state, references) {
+    const reference = references.find((r) => r.name === 'indexPatternLoad.id');
```

Comment:
In my experience, it's better if we can avoid ambiguous argument names like `r` in this case. It can improve the functional code readability.

```suggestion
    const reference = references.find(({ name }) => name === 'indexPatternLoad.id');
```